### DISCUSSION TITLE
Video files with a Alpha Channel

### DISCUSSION THREAD

**Discussion by jjouppi** [ffmpeginteropx/FFmpegInteropX#312] [General]
Hey ffmpeginteropx folks!

Wondering about handling alpha channels for video files such as ProRes 4444 with alpha? Is it at all possible to display transparency when video is playing back? Right now I can load such files but transparency is just rendered as black.

Also when using framegrabber is it possible to retain the alpha channel and encode a PNG or BMP with transparency? I use the frame grabber for an annotation feature in a media player and also am building a GIF exporter and it'd be great to retain transparency in both instances.

Thanks for all your work on this!

**Comment by lukasf** [comment]
Hi @jjouppi,

we have some support for alpha channel, but it does not work with every image format. I have a FLV video where alpha works fine, but also a ProRes file where alpha is not being used. Generally, you must allow BGRA8 and disable all other output formats:

```
            Config.VideoOutputAllowBgra8 = true;
            Config.VideoOutputAllowIyuv = false;
            Config.VideoOutputAllowNv12 = false;
```

Then set the background of the top level grid in the samples to blue, so you can see if alpha is in use or not. One issue is that black bars on top and bottom are always black, even if the video has transparency. There does not seem to be a way in MediaPlayerElement to disable or change that color. You could only try to work around this by sizing the MediaPlayerElement so that there are no bars.

The frame grabber currently drops the alpha channel, but I just created a PR #313 to allow exporting as PNG with alpha channel.

**Reply by jjouppi** [reply]
Hey Lukas!

Oh wonderful! I now have ProRes 4444 with alpha video playing back with transparency! Yeah black bars from MediaPlayerElement are a bit of a bummer visually but at least in my use case it has some practicality as it visualizes the dimensions of the video being reviewed. Really appreciate the help.

And thanks so much for making that update! Really going to simplify my annotation feature.

Wishing you all the best!

**Reply by lukasf** [reply]
Thanks for the feedback, glad to hear it's working for you!
